tests/test_AggregateAquiferData.cpp

```diff
@@ -0,0 +1,817 @@
+        };
+    }
+
+    Opm::EclipseGrid createGrid()
+    {
+        auto grid = Opm::EclipseGrid { 20, 5, 10, 5.0, 4.0, 0.2 };
+
+        auto actnum = std::vector<int>(grid.getCartesianSize(), 1);
+
+        actnum[grid.getGlobalIndex( 1, 1, 1)] = 0;
+        actnum[grid.getGlobalIndex( 2, 1, 1)] = 0;
+        actnum[grid.getGlobalIndex(19, 1, 6)] = 0;
+        actnum[grid.getGlobalIndex(19, 2, 6)] = 0;
+
+        grid.resetACTNUM(actnum);
+
+        return grid;
+    }
+
+    Opm::RestartIO::InteHEAD::AquiferDims syntheticAquiferDimensions()
+    {
+        auto aqDims = Opm::RestartIO::InteHEAD::AquiferDims{};
+
+        aqDims.numAquifers = 4; // 1, 2, 4, 6
+        aqDims.maxNumAquifers = 10; // >= 6
+        aqDims.maxNumAquiferConn = 5;       // >= 3
+        aqDims.maxNumActiveAquiferConn = 3; // ID = 1
+        aqDims.maxAquiferID = 6;
+
+        return aqDims;
+    }
+
+    Opm::RestartIO::InteHEAD::AquiferDims parseAquiferDimensions()
+    {
+        const auto deck = Opm::Parser{}.parseString(R"(RUNSPEC
+DIMENS
+  20 5 10 /
+
+AQUDIMS
+-- MXNAQN   MXNAQC   NIFTBL  NRIFTB   NANAQU    NNCAMAX
+    1*       1*        5       100      5         1000 /
+
+GRID
+
+DXV
+  20*100.0 /
+
+DYV
+  5*50.0 /
+
+DZV
+  10*10.0 /
+
+DEPTHZ
+  126*2000.0 /
+
+PORO
+  1000*0.25 /
+
+EQUALS
+  'PORO' 0.0 2 3 2 2 2 2 /
+  'PORO' 0.0 20 20 2 3 7 7 /
+/
+
+SOLUTION
+
+AQUFETP
+-- Aqu        depth     Pr      vol        Comp         PI        PVTW
+   2           2000.0   250.0   5.0E+10   1.5312E-4    495         2     /
+   4           2000.0   250.0   2.0E+10   1.5312E-4    910         1     /
+/
+
+AQUANCON
+-- Aq#     I1   I2    J1   J2   K1  K2  FACE
+    2      20   20     1    5    7   9   'I+'  1*   1.0 NO /
+    4      20   20     1    5    5   6   'I+'  1.0  1.0 NO /
+/
+
+)");
+
+        const auto es = Opm::EclipseState{ deck };
+
+        return Opm::RestartIO::inferAquiferDimensions(es);
+    }
+
+    Opm::SummaryState sim_state()
+    {
+        auto state = Opm::SummaryState{Opm::TimeService::now()};
+
+        state.update("AAQP:1", 123.456);
+        state.update("AAQR:1", 234.567);
+        state.update("AAQT:1", 3456.789);
+        state.update("AAQPD:1", 4.567);
+        state.update("AAQTD:1", 50.607);
+
+        state.update("AAQP:2", 121.212);
+        state.update("AAQR:2", 222.333);
+        state.update("AAQT:2", 333.444);
+
+        state.update("AAQP:4", 555.444);
+        state.update("AAQR:4", 333.222);
+        state.update("AAQT:4", 222.111);
+
+        state.update("AAQP:6", 456.123);
+        state.update("AAQR:6", 34.567);
+        state.update("AAQT:6", 4444.5555);
+        state.update("AAQPD:6", 50.706);
+        state.update("AAQTD:6", 100.321);
+
+        return state;
+    }
+
+    template <class Coll1, class Coll2>
+    void check_is_close(const Coll1& coll1, const Coll2& coll2, const double tol)
+    {
+        BOOST_REQUIRE_EQUAL(std::distance(std::begin(coll1), std::end(coll1)),
```

Comment:
Ehhhh - are there relevant containers which don't have `size()`?

Reply:
> are there relevant containers which don't have `size()`?

The most obvious example would be built-in arrays.  We can use [`std::size()`](https://en.cppreference.com/w/cpp/iterator/size) for those however, so I've pushed an update to do that.